server/src/main/java/io/druid/server/coordination/SegmentChangeRequestDrop.java

```diff
@@ -62,4 +62,29 @@ public String toString()
            "segment=" + segment +
            '}';
   }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    SegmentChangeRequestDrop that = (SegmentChangeRequestDrop) o;
+
+    if (!segment.equals(that.segment)) {
```

Comment:
This is all IntelliJ auto-generated code as far as I can tell. As long as we agree on whether subclasses should be accepted for `equals`, which fields to account for, and which fields can be null, I think it's reasonable to trust the generated code. We typically don't use getters anywhere else for equals methods, so unless there's a good reason to, I would leave it as is for consistency.